projects/packages/woocommerce-analytics/src/class-woo-analytics-trait.php

```diff
@@ -828,7 +835,7 @@ public function get_session_cookie() {
 	 * @return string
 	 */
 	public function get_current_url() {
-		return sanitize_url( wp_unslash( ( empty( $_SERVER['HTTPS'] ) ? 'http' : 'https' ) . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]" ) ); // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotValidatedNotSanitized -- actually escaped with sanitize_url.
+		return home_url( add_query_arg( null, null ) );
```

Comment:
I replaced usage of the $_SERVER variable with home_url() to retrieve the current URL, as home_url() should generally be more dependable.